### PR TITLE
[RFC] Cleanup whitelist

### PR DESCRIPTION
**Pull request by bezin**
Moved general terms to default.txt. Highly opioniated and probably battlesome.

### CODE REVIEW

**Comment by aschempp** [comment]
Great find!

Can we also fix the stuff that is misspelled?
 - Instead of `eMail` and `Email` we should only allow `E-Mail`. Not sure if the linter can do dashes @Toflar?
 - We should only allow one version of things, like `ChangeLanguage` (product name) would be correct but `Changelanguage` is not.

**Comment by Toflar** [comment]
>     * Not sure if the linter can do dashes @Toflar?

No idea, I just use `aspell`.

File: linter/whitelists/default.txt [hunk 1]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
-Slack
\ No newline at end of file
+Slack
+Catalog
+Manager
+Backend
+Bing
+Bundle
+Bundles
+Changelanguage
```
Comment (Toflar) [inline linter/whitelists/default.txt]: `Changelanguage` is worng, this should be fixed in the package description :)

File: linter/whitelists/default.txt [hunk 2]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+Templates
+Templating
+Themes
+Theme
+Usability
+WhatsApp
+Widget
+Widgets
+Wizard
+backend
```
Comment (Toflar) [inline linter/whitelists/default.txt]: `backend` is wrong in my opinion. In German, I would use `Backend` and in English it's `back end` or `back-end`. But never `backend`.

File: linter/whitelists/default.txt [hunk 3]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+contao
```
Comment (Toflar) [inline linter/whitelists/default.txt]: That shouldn't be allowed, it's always `Contao`.

File: linter/whitelists/default.txt [hunk 4]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+estao
+geocoding
+Frontend
```
Comment (Toflar) [inline linter/whitelists/default.txt]: Same rules as my comment on `Backend`.

File: linter/whitelists/default.txt [hunk 5]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+Framework
+Footer
+theme
+symfony
```
Comment (Toflar) [inline linter/whitelists/default.txt]: `Symfony` is a name, always uppercase.

File: linter/whitelists/default.txt [hunk 6]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+php
+pdf
+rss
+jQuery
+javascript
```
Comment (Toflar) [inline linter/whitelists/default.txt]: The one and only correct way is `JavaScript`.

File: linter/whitelists/default.txt [hunk 7]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+eCommerce
+eMail
+XING
+XLIFF
+XLS
+Xing
```
Comment (Toflar) [inline linter/whitelists/default.txt]: As far as I can see, `XING` is the only correct way.

File: linter/whitelists/default.txt [hunk 8]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+OpenStreetMap
+OpenGraph
+OpenLayers
+Overpass
+Nominatim
+OSM
+JPEG
+JPG
+JavaScript
+Javascript
```
Comment (Toflar) [inline linter/whitelists/default.txt]: That should be removed, only `JavaScript` is correct.

File: linter/whitelists/default.txt [hunk 9]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+Community
+Composer
+CMS
+CSS
+CSV
+Checkbox
+App
+Android
+Analytics
+excel
```
Comment (Toflar) [inline linter/whitelists/default.txt]: `Excel`

File: linter/whitelists/default.txt [hunk 10]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+Overpass
+Stylesheet
+UI
+WMS
+adwords
+ajax
+amp
+app
+euf
+faq
```
Comment (Toflar) [inline linter/whitelists/default.txt]: `FAQ`, right?

File: linter/whitelists/default.txt [hunk 11]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+frontend
+gis
+google
```
Comment (Toflar) [inline linter/whitelists/default.txt]: `Google`?

File: linter/whitelists/default.txt [hunk 12]
```diff
@@ -20,4 +21,119 @@ TinyPNG
...
+googlemaps
```
Comment (Toflar) [inline linter/whitelists/default.txt]: `Google Maps`

**Comment by bezin** [comment]
I think some lower case variants of some terms come from package names in the description. Could it be that aspell splits terms with underscore and spell-checks each part individually. Thats probably why dlh and googlemaps are whitelisted. 

meta\trilobit-gmbh\contao-monitoring-client-sensor-googlemaps\en.yml
"The module only works with the google maps extension dlh_googlemaps"

Same might apply to "contao".

**Comment by Toflar** [comment]
I see. I don't see any way to solve that then. Let's face it, the spell checker is not perfect but it can help us to find some obvious spelling issues so that's a good thing 😄 Where exactly we manage the whitelist doesn't really matter. Do you want to check my review comments and change something about them before I merge?

File: linter/whitelists/default.txt (hunk 3 above)
Comment (bezin) [reply linter/whitelists/default.txt]: Is used in an extension name

File: linter/whitelists/default.txt (hunk 5 above)
Comment (bezin) [reply linter/whitelists/default.txt]: Corrected, but might be used in a bundle name as well

File: linter/whitelists/default.txt (hunk 9 above)
Comment (bezin) [reply linter/whitelists/default.txt]: Is used in a package name

**Comment by bezin** [comment]
@toflar I fixed some things you mentioned. Some are not resolvable, because the terms are used in package names. 

I agree with you on front-end and back-end and yet I am somebody constantly writing backend and frontend in English myself 8-) Any other opinions on that?

**Comment by xchs** [comment]
Leo actually writes **back end** and **front end** everywhere. That is why we should stick to it as well.

**Comment by aschempp** [comment]
I don't think we should lint the package name, but only the package title?

**Comment by Toflar** [comment]
@bezin so it's already 3 people that voted for `back end` and `front end` 😄 



> I don't think we should lint the package name, but only the package title?

Yeah, we only do lint the title but still, most of the translations just write the name of the package in the title 😄

**Comment by aschempp** [comment]
> Yeah, we only do lint the title but still, most of the translations just write the name of the package in the title 😄

yeah, they write the "marketing name", which might be "Google Maps Module for Contao Monitoring", but they should not write `contao-monitoring-client-sensor-googlemaps does XY` imho.

**Comment by Toflar** [comment]
That's true but that's what we currently have 😄

**Comment by bezin** [comment]
Yeah, I am sold on front end and back end. Anyhow, what about all that German terms in the English and Czech language files? ¯\_(ツ)_/¯

**Comment by Toflar** [comment]
Noooo 😄 In German `Backend` and `Frontend` is fine 🙈

**Comment by bezin** [comment]
It is Backend and Frontend in German and back end and front end in English now ;-)

**Comment by Toflar** [comment]
Let's leave it at that, it's maybe not perfect but already a major improvement! Thank you so much ❤️

**Pull request merged**